Microsoft.Azure.CosmosRepository/src/Providers/DefaultCosmosClientProvider.cs

```diff
@@ -1,43 +1,82 @@
-﻿// Copyright (c) IEvangelist. All rights reserved.
-// Licensed under the MIT License.
-
-using System;
-using System.Threading.Tasks;
-using Microsoft.Azure.Cosmos;
-using Microsoft.Azure.CosmosRepository.Options;
-using Microsoft.Extensions.Options;
+﻿// Copyright (c) IEvangelist. All rights reserved. Licensed under the MIT License.
```

Comment:
This entire file can be reverted to its original version, the dispose pattern isn't needed here since all we're doing is cascading a call to dispose. No reason to maintain state of dispose invocations, provide finalizer, or GC calls, etc. - keep it simple.